### PR TITLE
Pass correct environment to tools.

### PR DESCRIPTION
**Pull request by grp**
- Include all necessary paths in tool search paths.
 - Set `PATH` to match the paths searched for tools.
 - Set `DEVELOPER_DIR` for all invocations to stay self-contained.
 - Pass along process environment to tools rather than clearing.

My version of #169. @onhachoe, would be great if you could test with the case you were seeing `PATH` being important.

### CODE REVIEW

File: Libraries/xcexecution/Sources/NinjaExecutor.cpp [hunk 1]
```diff
@@ -806,9 +806,9 @@ buildInvocation(
     }
 
     /*
-     * Build the invocation environment. To set the environment, we use standard shell syntax.
-     * Use `env` to avoid Bash-specific limitations on environment variables. Specifically, some
-     * versions of Bash don't allow setting "UID". Pass -i to clear out the environment.
+     * Build the invocation environment. To set the environment, we use standard shell tools:
+     * `env` to avoid Bash-specific limitations on environment variables (some versions of Bash
+     * don't allow setting "UID"). Intentionally add to, not replace, the process environment.
```
Comment (onhachoe) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: Is there a reason for changing the behavior?

Comment (grp) [inline Libraries/xcexecution/Sources/NinjaExecutor.cpp]: The old behavior was wrong, AFAICT.

**Comment by onhachoe** [comment]
Wait, but how about other tools than ibtool actool? shouldn't we add support for scripts as well?

**Comment by grp** [comment]
Scripts pass all build settings as environment variables, including `PATH`.

**Pull request merged**